src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -159,6 +168,25 @@ public override object Id
             }
         }
 
+        /// <summary>
+        /// Gets the underlying type of the component that was loaded if it was wrapped at load time
+        /// </summary>
+        /// <remarks>
+        /// Types returned from e.g. <see cref="GetGeneratorsForAllLanguages()"/> or <see cref="GetGenerators(string)"/> may be wrapped
+        /// in an adaptor for backwards compatibility reasons. This method allows the caller to get at the underlying type
+        /// without having to know if it was wrapped or not.
+        /// </remarks>
+        /// <param name="loadedType">The type that was returned from the loader</param>
+        /// <returns>The underlying type if wrapped, or the passed in type if not</returns>
+        public static Type UnwrapTypeIfRequired(Type loadedType)
```

Comment:
So does this need to be on Analyzer**File**Reference or somewhere else? Because this same trick has to be used for anybody creating analyzer references that aren't file references, right?